chain/consensus/mir/config.go

```diff
@@ -44,17 +58,37 @@ func NewConfig(
 	dbPath string,
 	initCheck *checkpoint.StableCheckpoint,
 	checkpointRepo string,
-	segmentLength int,
+	segmentLength, configOffset int,
+	maxProposeDelay time.Duration,
 	rpcServerURL string,
+	membershipType string,
+
 ) *Config {
+	if membershipType == "" {
+		membershipType = DefaultMembershipType
+	}
```

Comment:
Here it would be nice to check if the received value is an allowed one and complain if it is not.
Using an (emulated) enum type (as mentioned in an earlier comment) would help here.

Reply:
fixed

Reply:
Fixed. Now the validation function will panic if the membership is unknown.

Reply:
Where does it panic? It seems to me that it still doesn't, but uses the default value. Even the latest version of the code still has
```go
	if !membership.IsSourceValid(membershipSource) {
		membershipSource = DefaultMembershipSource
	}
```
Where `membership.IsSourceValid(membershipSource)` never panics.

Reply:
Thank you, now fixed.